main.go

```diff
@@ -2,137 +2,182 @@ package main
-		return fmt.Errorf("event does not contain check")
-	}
+	emailSubjectTemplate = "Sensu Alert - {{.Entity.Name}}/{{.Check.Name}}: {{.Check.State}}"
+	emailBodyTemplate    = "{{.Check.Output}}"
 
-	sendMailError := sendEmail(event)
-	if sendMailError != nil {
-		return fmt.Errorf("failed to send email: %s", sendMailError)
+	emailConfigOptions = []*sensu.PluginConfigOption{
+		{
+			Path:      smtpHost,
+			Argument:  smtpHost,
+			Shorthand: "s",
+			Default:   "",
+			Usage:     "The SMTP host to use to send to send email",
+			Value:     &config.SmtpHost,
+		},
+		{
+			Path:      smtpUsername,
+			Env:       "SMTP_USERNAME",
+			Argument:  smtpUsername,
+			Shorthand: "u",
+			Default:   "",
+			Usage:     "The SMTP username, if not in env SMTP_USERNAME",
+			Value:     &config.SmtpUsername,
+		},
+		{
+			Path:      smtpPassword,
+			Env:       "SMTP_PASSWORD",
+			Argument:  smtpPassword,
+			Shorthand: "p",
+			Default:   "",
+			Usage:     "The SMTP password, if not in env SMTP_PASSWORD",
+			Value:     &config.SmtpPassword,
+		},
+		{
+			Path:      smtpPort,
+			Argument:  smtpPort,
+			Shorthand: "P",
+			Default:   uint64(defaultSmtpPort),
+			Usage:     "The SMTP server port",
+			Value:     &config.SmtpPort,
+		},
+		{
+			Path:      toEmail,
+			Argument:  toEmail,
+			Shorthand: "t",
+			Default:   "",
+			Usage:     "The 'to' email address",
+			Value:     &config.ToEmail,
+		},
+		{
+			Path:      fromEmail,
+			Argument:  fromEmail,
+			Shorthand: "f",
+			Default:   "",
+			Usage:     "The 'from' email address",
+			Value:     &config.FromEmail,
+		},
+		{
+			Path:      insecure,
+			Argument:  insecure,
+			Shorthand: "i",
+			Default:   false,
+			Usage:     "Use an insecure connection (unauthenticated on port 25)",
+			Value:     &config.Insecure,
+		},
+		{
+			Path:      hookout,
+			Argument:  hookout,
+			Shorthand: "H",
+			Default:   false,
+			Usage:     "Include output from check hook(s)",
+			Value:     &config.Hookout,
+		},
+		{
+			Path:      bodyTemplateFile,
+			Argument:  bodyTemplateFile,
+			Shorthand: "T",
+			Default:   "",
+			Usage:     "A template file to use for the body",
+			Value:     &config.BodyTemplateFile,
+		},
 	}
+)
 
-	return nil
+func main() {
+	goHandler := sensu.NewGoHandler(&config.PluginConfig, emailConfigOptions, checkArgs, sendEmail)
+	goHandler.Execute()
 }
 
-func checkArgs() error {
-	if len(smtpHost) == 0 {
+func checkArgs(_ *corev2.Event) error {
+	if len(config.SmtpHost) == 0 {
 		return errors.New("missing smtp host")
 	}
-	if len(toEmail) == 0 {
+	if len(config.ToEmail) == 0 {
 		return errors.New("missing destination email address")
 	}
-	if !insecure {
-		if len(smtpUsername) == 0 {
+	if !config.Insecure {
+		if len(config.SmtpUsername) == 0 {
 			return errors.New("smtp username is empty")
 		}
-		if len(smtpPassword) == 0 {
+		if len(config.SmtpPassword) == 0 {
 			return errors.New("smtp password is empty")
 		}
 	} else {
-		smtpPort = 25
+		config.SmtpPort = 25
```

Comment:
Since `SmtpPort` is defined as a `uint64` and not a `uint16` due to lack of support in `sensu-plugins-go-library` you might want to add an extra check to make sure it's within the 16 bit range.

Reply:
Makes sense. Fixed it in latest commit.